cli/daemon/run/run.go

```diff
@@ -279,7 +279,7 @@ func (r *Run) buildAndStart(ctx context.Context, tracker *optracker.OpTracker) e
 	}
 
 	var build *compiler.Result
-	jobs.Go("Compiling application source code", false, 0, func(ctx context.Context) (err error) {
+	jobs.Go("Compiling Encore application", true, 0, func(ctx context.Context) (err error) {
```

Comment:
I don't this makes sense as the compiler.Build process within the job already fully tracks errors, this way we're going to end up with nested tracked jobs... maybe that's fine - but I'm not sure that's the root cause here.

I suspect it might be multiple rounds of panic/recovery hiding the error